### PR TITLE
[cuDNN][SDPA] Don't route to cuDNN SDPA when dropout probability is not a multiple of 1/16

### PR DESCRIPTION
**Pull request by eqy**
certain cuDNN SDPA kernels only have a dropout probability resolution of 1/16, and this implicit rounding is a gotcha for users/tests

### CODE REVIEW

File: test/test_transformers.py [hunk 1]
```diff
@@ -2960,6 +2960,14 @@ def test_cudnn_attention_seqlen1_dropout_heuristic(self):
             out = torch.nn.functional.scaled_dot_product_attention(q, q, q, dropout_p=0.5)
             out.backward(grad)
 
+    @unittest.skipIf(not PLATFORM_SUPPORTS_CUDNN_ATTENTION, "cudnn Attention is not supported on this system")
+    def test_cudnn_attention_low_dropout(self):
+        q = torch.randn(2, 8, 128, 128, dtype=torch.half, device='cuda')
```
Comment (drisspg) [inline test/test_transformers.py:2977]: can we asssert not equal and xfail?

Comment (drisspg) [reply test/test_transformers.py:2977]: idk how reliable that test will be

**Review by ngimel — APPROVED** [review]
